### PR TITLE
Preconditions

### PR DESCRIPTION
**Pull request by juandbc**
## Summary
* Add guava dependency
* Add apache commons lang3 dependency
* Add checkNotNull precondition to model classes setters
* Add error message constants for pre-conditions 
* Improve sign up precondition validations

## Trello Card
https://trello.com/c/xqOXL9pq/29-agregar-validaci%C3%B3n-de-precondiciones-a-los-setters

### CODE REVIEW

File: src/main/java/wolox/training/models/Book.java [hunk 1]
```diff
@@ -76,63 +78,63 @@ public String getAuthor() {
     }
 
     public void setAuthor(String author) {
-        this.author = author;
+        this.author = checkNotNull(author);
```
Comment (jeymarsonwolox) [inline src/main/java/wolox/training/models/Book.java]: check blank string

File: src/main/java/wolox/training/models/Book.java [hunk 2]
```diff
@@ -76,63 +78,63 @@ public String getAuthor() {
...
     }
 
     public String getImage() {
         return image;
     }
 
     public void setImage(String image) {
-        this.image = image;
+        this.image = checkNotNull(image);
```
Comment (jeymarsonwolox) [inline src/main/java/wolox/training/models/Book.java]: same

**Review by jeymarsonwolox — CHANGES_REQUESTED** [review]
Great Job!! Validate nulls and blanks in strings

**Review by jeymarsonwolox — DISMISSED** [review]

File: src/main/java/wolox/training/models/User.java [hunk 3]
```diff
@@ -76,23 +81,23 @@ public LocalDate getBirthdate() {
     }
 
     public void setBirthdate(LocalDate birthdate) {
-        this.birthdate = birthdate;
+        this.birthdate = checkNotNull(birthdate);
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/models/User.java]: After checking if it is null, you can check if the date is after the current one (the date of birth must always be earlier).

File: src/main/java/wolox/training/models/Book.java [hunk 4]
```diff
@@ -116,6 +125,7 @@ public String getYear() {
     }
 
     public void setYear(String year) {
+        checkArgument(!Strings.isNullOrEmpty(year));
         this.year = year;
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/models/Book.java:143]: After checking if it is null or empty, you can check if the content of the year field is a number.

You can use the StringUtils.isNumeric method. [import org.apache.commons.lang3.StringUtils;]

File: src/main/java/wolox/training/models/Book.java [hunk 5]
```diff
@@ -132,6 +143,7 @@ public String getIsbn() {
     }
 
     public void setIsbn(String isbn) {
+        checkArgument(!Strings.isNullOrEmpty(isbn));
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/models/Book.java]: After checking if it is null or empty, you can check if the content of the isbn field is a number.

File: src/main/java/wolox/training/models/Book.java [hunk 6]
```diff
@@ -124,6 +134,7 @@ public String getPages() {
     }
 
     public void setPages(String pages) {
+        checkArgument(!Strings.isNullOrEmpty(pages));
         this.pages = pages;
```
Comment (Gianni-Nicolas) [inline src/main/java/wolox/training/models/Book.java:154]: After checking if it is null or empty, you can check if the content of the page field is greater than 0.

**Review by Gianni-Nicolas — CHANGES_REQUESTED** [review]
Great Job 👍

**Review by jeymarsonwolox — DISMISSED** [review]

**Review by Gianni-Nicolas — APPROVED** [review]

**Review by jeymarsonwolox — APPROVED** [review]

**Pull request merged**